### ISSUE TITLE
Error in repeat_linter with unexpected object spacing

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2828]
```r
lint("       while\n(TRUE) {}", repeat_linter())
# Error in `lint()`:
# ! Linter `linter()` failed in /tmp/RtmpIOiePB/file351e149a61c9c:
# Caused by error in `validate_lint_object()` at lintr/R/lint.R:382:3:
# ! `ranges` must satisfy 0 <= range[1L] <= range[2L] <= 13 (nchar(line) + 1).
# Run `rlang::last_trace()` to see where the error occurred.
```
